Added setSize()

daamsie opened the pull request:
#5 This adds the ability to set a size for the generated map image

I'm using the GLOBAL scope to store the size. Maybe there's a better way, but this seems to work.

jieter commented:
Hmm, unit tests fail, not related to PR.

I think I don't like the idea of using `GLOBAL`: having a `setSize()` on a map instance leads to expectations about that instance. Will be suprising at least. Why not keep them as class members? Can you also update the `README.md` as it documents the fixed size...

daamsie commented:
The tests passed on my local setup, except for the last one sometimes did and sometimes didn't as it seemed to take too long sometimes.

daamsie commented:
Ok, I've updated and removed the GLOBAL variable. Internally Leaflet uses _size to store the size of the map, so setSize is now directly adjusting that instead. getSize() is now also tweaked to be the same as the original function, except for not referencing the container's width and height.

Also updated the README this time.

jieter commented on index.js:
```diff
@@ -26,7 +26,13 @@ if (!GLOBAL.L) {
+		if (!this._size || this._sizeChanged) {
+			this._size = new L.Point(
+				1024,
+				1024);
```
Why is this not in one line?

jieter commented on test/test.js:
```diff
@@ -35,6 +34,14 @@ describe('Leaflet-headless', function () {
+			var size = map.getSize();
+			size.x.should.equal(800);
+			size.y.should.equal(600);
+		});
```
Maybe another test actually changing the size on a map instance might be a good addition? Another comma here.

daamsie replied:
After setView() the size is set to the default 1024x1024 . So this does actually change the size.

jieter commented:
I've added some line notes.

Style issues should be caught by eslint in CI, which obviously doesn't work now. I'll look into fixing that.

jieter commented:
I just fixed the build in master, you can use that if you rebase.

jieter commented:
Looks good now, can you squash the commits into one?

daamsie commented:
I did this (learned something new there), but it does squash your commit into it as well. Not sure what happens to that when you merge the PR in. Hopefully it all works.

jieter commented:
Merged. Seems to be no problem. 

I'll release to npm later today.

jieter commented:
Just [released `v0.1.1`](https://www.npmjs.com/package/leaflet-headless), thanks for the pull!

Would have been cleaner if you rebased on master (`git fetch` / `git rebase upstream master`), maybe next time ;)

daamsie commented:
No worries. Glad to have helped just a little bit : )
